src/Pact/Types/Term.hs

```diff
@@ -253,6 +254,29 @@ instance Semigroup Gas where
 instance Monoid Gas where
   mempty = 0
 
+newtype MicroGas
+  = MicroGas Int64
+  deriving (Eq, Ord, ToJSON, FromJSON, NFData, Generic)
```

Comment:
Copied over most instances from `Gas` but I don't believe we need `ToJSON` or `FromJSON`.

Notably though: no `Num` instance. We must use monoid concat to increase gas, which is what we want: gas shouldn't ever decrease after being charged, and negative gas isn't something we want.

Reply:
Deleting the JSON instances if possible would be nice. Units are confusing and type class resolution is confusing, so forcing manual serialization might reduce the footgun risk.